spring-security/src/main/java/com/sap/cloud/security/spring/config/OAuth2ServiceConfigurationProperties.java

```diff
@@ -0,0 +1,151 @@
+package com.sap.cloud.security.spring.config;
+
+import com.sap.cloud.security.config.OAuth2ServiceConfiguration;
+import com.sap.cloud.security.config.OAuth2ServiceConfigurationBuilder;
+import com.sap.cloud.security.config.Service;
+import com.sap.cloud.security.config.cf.CFConstants;
+
+import java.net.URI;
+import java.util.Map;
+
+/**
+ * PoJo of Spring configuration properties.
+ * It implements the {@link OAuth2ServiceConfiguration} and enhances it with
+ * <ul>
+ *   <li>setters</li>
+ *   <li>Xsuaa specific setters to map Xsuaa specific configuration such as {@code uaadomain}, {@code xsappname}</li>
+ * </ul>
+ * You can define your own prefix when declaring {@code @ConfigurationProperties("xsuaa")}.
+ * This then needs to be mapped to {@code $vcap.services.<your service instance name>.credentials}
+ * as part of your application properties. <br>
+ *
+ * Alternatively, its possible to directly map the properties to {@code VCAP_SERVICES}
+ * system environment variable: <br>
+ * {@code @ConfigurationProperties("vcap.services.<your service instance name>.credentials")}
+ */
+public class OAuth2ServiceConfigurationProperties implements OAuth2ServiceConfiguration {
```

Comment:
Maybe you remember that in CAP we have an abstraction of OSB API Service Bindings. We can meanwhile fill those either from default-env.json, VCAP_SERVICES or from some file-system structure on Kubernetes and with that make the rest of the CAP framework mostly CF/K8s independent.
With the current XSUAA Spring Library we implemented our own XsuaaPropertySourceFactory to connect our ServiceBindings abstractions with your library. I think we should check that this also works with this new configuration approach.

Reply:
Hi @beckermarc thanks for the prompt feedback!
Eager to learn how you did the integration stuff with your own XsuaaPropertySourceFactory implementation. As of now, i have not chosen that in order to support multiple xsuaa bindings, but maybe you have a solution for that too?

Reply:
created a BLI: SECAUTH-1269

Reply:
Hi @beckermarc 

@PropertySource integration is supported now, as documented here:
https://github.com/SAP/cloud-security-xsuaa-integration/tree/hybrid-jwt-decoder/spring-security#setup-spring-security-oauth-20-resource-server